### PR TITLE
manage collaborators view

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1345

Schermata 2022-05-30 alle 14 09 45

### CODE REVIEW

File: client/store/userManagement/actions/getCountryUsers.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import { CountryIso } from '@meta/area'
+import { User } from '@meta/user'
+
+type Params = {
+  countryIso: CountryIso
+}
+
+export const getCountryUsers = createAsyncThunk<Array<User>, Params>(
+  'userManagement/getCountryUsers',
```
Comment (sorja) [inline client/store/userManagement/actions/getCountryUsers.ts:13]: should this be `usermanagement/get/countryUsers`

File: server/api/user/index.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+import { getByCountry } from './getByCountry'
+import { getInvitation } from './getInvitation'
 import { getProfilePicture } from './getProfilePicture'
 
 export const UserApi = {
   init: (express: Express): void => {
     express.get(ApiEndPoint.User.getByInvitation(), getInvitation)
     express.get(ApiEndPoint.User.acceptInvitation(), getAcceptInvitation)
     express.get(ApiEndPoint.User.getProfilePicture(), getProfilePicture)
+    express.get(ApiEndPoint.User.getByCountry(), getByCountry)
```
Comment (sorja) [inline server/api/user/index.ts:14]: Should this have view auth middleware?

Comment (rfontanarosa) [reply server/api/user/index.ts:14]: probably...

File: common/api/endpoint/ApiEndPoint.ts [hunk 3]
```diff
@@ -88,5 +88,6 @@ export const ApiEndPoint = {
     getByInvitation: (uuid = ':uuid') => apiPath('user', 'invitation', uuid),
     acceptInvitation: (uuid = ':uuid') => apiPath('user', 'invitation', 'accept', uuid),
     getProfilePicture: (id = ':id') => apiPath('user', 'profilePicture', id),
+    getByCountry: (countryIso = ':countryIso') => apiPath('user', countryIso),
```
[the remark this answers is not in the document]
Comment (minotogna) [reply common/api/endpoint/ApiEndPoint.ts:91]: this should just be many(). and be ```/users```. 
also the objects to be renamed to Users and other endpoints should start with /users

**Review by minotogna — CHANGES_REQUESTED** [review]
repository is missing.

File: common/api/endpoint/ApiEndPoint.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (rfontanarosa) [reply common/api/endpoint/ApiEndPoint.ts:91]: what do you mean with objects and which other endpoint should starts with users? all the others return only one user.

File: server/api/user/index.ts [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+import { getInvitation } from './getInvitation'
+import { getMany } from './getMany'
 import { getProfilePicture } from './getProfilePicture'
 
 export const UserApi = {
   init: (express: Express): void => {
     express.get(ApiEndPoint.User.getByInvitation(), getInvitation)
     express.get(ApiEndPoint.User.acceptInvitation(), getAcceptInvitation)
     express.get(ApiEndPoint.User.getProfilePicture(), getProfilePicture)
+    express.get(ApiEndPoint.User.many(), getMany)
```
Comment (minotogna) [inline server/api/user/index.ts:14]: We do need an auth middleware 

Only users that have a role in that county can see the users .

Do you want to add it in this pr or next pr ?

Comment (rfontanarosa) [reply server/api/user/index.ts:14]: nope, if it's only for the country it's easy to do

**Review by minotogna — APPROVED** [review]

**Pull request merged**